### ISSUE TITLE
WebContent crash or weird JS console behavior after a JS exception during page load

### ISSUE THREAD

**Issue by TSultanov** [LadybirdBrowser/ladybird#782]
I found a way to reliably crash WebContent with `ConnectionFromClient(0x000000015960a0d0) (id=1) misbehaved (Requested non-existent console message index.), disconnecting.` error message or cause weird console behavior.

Steps to reproduce:
1. Create the following HTML file
```html
<html>
    <head>
        <title>WebContent Crash</title>
    </head>
    <body>
        <div>
            <select id="test-element">
                <option>A</option>
                <option>B</option>
            </select>
        </div>
        <script>
            throw 123;
        </script>
    </body>
</html>
```
2. Open the file in Ladybird
3. Open the Inspector window
4. Run `console.log(123)` in the JS Console
5. Browser will report that the tab has crashed

The above steps reliably cause a crash on macOS with the Debug build configuration and with the native chrome and crash or weird console behavior (results of evaluations not showing up, or results of a previous evaluation showing up) with the Qt chrome.

Edit: updated repro to not depend on a confounding bug in HTMLSelectElement which is fixed now

**Comment by sideshowbarker** [comment]
I was able to reproduce this. The Ladybird log output looks like this:
```
110070.817 WebContent(48967): Unhandled JavaScript exception: [TypeError] Cannot access property "selected" on undefined object "selector.selectedOptions[0]"
110070.817 WebContent(48967):     at file:///opt/ladybird/2TEST.html:3:25

110097.793 WebContent(48967): (js log) 123
110097.793 Ladybird(48964): Notified about console message we're already aware of
110097.794 WebContent(48967): ConnectionFromClient(0x000000015b7124e0) (id=1) misbehaved (Requested non-existent console message index.), disconnecting.
GrMtlDepthStencil: 0x600002c68000 (2 refs)
GrMtlRenderPipeline: 0x15b8b1000 (14 refs)
GrMtlRenderPipeline: 0x15b990400 (7 refs)
GrMtlDepthStencil: 0x600002c55180 (2 refs)
GrMtlDepthStencil: 0x600002c54c30 (2 refs)
GrMtlRenderPipeline: 0x15b99a600 (4 refs)
GrMtlDepthStencil: 0x600002c55400 (2 refs)
GrMtlRenderPipeline: 0x15d87aa00 (4 refs)
GrMtlSampler: 0x600003e7d360 (1 refs)
GrMtlRenderPipeline: 0x15d88c200 (18 refs)
GrMtlRenderPipeline: 0x15d00a000 (6 refs)
../src/1c27019d33-728a6df320.clean/src/gpu/ganesh/GrManagedResource.h:56: fatal error: "check(0 == fHashSet.count())"
110097.796 Ladybird(48964): WebContent process crashed!
```

**Comment by TSultanov** [comment]
Did some further debugging with `dbgln` sprinkled through the code. What's interesting is than when document contains the `<select>` element there is an additional call to `Document::initialize()`, which causes a new `WebContentConsoleClient` allocation (`0x0000000115d58448`), and the exception is reported into that new console client. And, when Inspector gets opened, it tries to retrieve the log from a different console client (`0x0000000115d582f0`).
But when the document is empty, i.e. I commented out the `<select>` tag, then the same `WebContentConsoleClient` is used in both cases (`0x00000001176782f0`).
### With `<select>` element present
```
5671184.945 WebContent(54375): Document(0x0000000115cd4040)::initialize(0x0000000115b30040)
5671184.945 WebContent(54375): PageClient(0x0000000115b18040)::initialize_js_console(0x0000000115cd4040): console_client = 0x0000000115d58040
5671184.946 WebContent(54375): Document(0x0000000115cd4040)::make_active()
5671184.946 WebContent(54375): PageClient(0x0000000115b18040)::page_did_change_active_document_in_top_level_browsing_context(0x0000000115cd4040): m_top_level_document_console_client = 0x0000000115d58040
5671185.010 WebContent(54375): Document(0x0000000115cd47e8)::initialize(0x0000000115b300b8)
5671185.010 WebContent(54375): PageClient(0x0000000115b18040)::initialize_js_console(0x0000000115cd47e8): console_client = 0x0000000115d58198
5671185.010 WebContent(54375): Document(0x0000000115cd47e8)::make_active()
5671185.010 WebContent(54375): PageClient(0x0000000115b18040)::page_did_change_active_document_in_top_level_browsing_context(0x0000000115cd47e8): m_top_level_document_console_client = 0x0000000115d58198
5671190.285 WebContent(54375): Document(0x000000011a134040)::initialize(0x0000000115b30130)
5671190.285 WebContent(54375): PageClient(0x0000000115b18040)::initialize_js_console(0x000000011a134040): console_client = 0x0000000115d582f0
5671190.298 WebContent(54375): Document(0x000000011a134040)::make_active()
5671190.298 WebContent(54375): PageClient(0x0000000115b18040)::page_did_change_active_document_in_top_level_browsing_context(0x000000011a134040): m_top_level_document_console_client = 0x0000000115d582f0
5671190.299 WebContent(54375): Document(0x000000011a264040)::initialize(0x0000000115b30130)
5671190.299 WebContent(54375): PageClient(0x0000000115b18040)::initialize_js_console(0x000000011a264040): console_client = 0x0000000115d58448
5671190.300 WebContent(54375): Unhandled JavaScript exception: aaa
5671190.300 WebContent(54375): Console(0x0000000115c4c120)::report_exception(): m_client = 0x0000000115d58448
5671190.300 WebContent(54375): WebContentConsoleClient(0x0000000115d58448)::print_html(<span style="color: red;">Uncaught [Error]: </span>aaa<br>)
5671190.300 WebContent(54375): WebContentConsoleClient(0x0000000115d58448)::print_html(): m_message_log.size() = 1
5671193.392 WebContent(54430): Document(0x000000011576c040)::initialize(0x00000001155c8040)
5671193.392 WebContent(54430): PageClient(0x00000001155b0040)::initialize_js_console(0x000000011576c040): console_client = 0x00000001157f0040
5671193.393 WebContent(54430): Document(0x000000011576c040)::make_active()
5671193.393 WebContent(54430): PageClient(0x00000001155b0040)::page_did_change_active_document_in_top_level_browsing_context(0x000000011576c040): m_top_level_document_console_client = 0x00000001157f0040
5671193.471 WebContent(54430): Document(0x000000011576c7e8)::initialize(0x00000001155c80b8)
5671193.471 WebContent(54430): PageClient(0x00000001155b0040)::initialize_js_console(0x000000011576c7e8): console_client = 0x00000001157f0198
5671193.473 WebContent(54430): Document(0x000000011576c7e8)::make_active()
5671193.473 WebContent(54430): PageClient(0x00000001155b0040)::page_did_change_active_document_in_top_level_browsing_context(0x000000011576c7e8): m_top_level_document_console_client = 0x00000001157f0198
5671193.489 WebContent(54375): PageClient(0x0000000115b18040)::js_console_request_messages(0): m_top_level_document_console_client = 0x0000000115d582f0
5671193.489 WebContent(54375): WebContentConsoleClient(0x0000000115d582f0)::send_messages(0)
5671193.489 WebContent(54375): WebContentConsoleClient(0x0000000115d582f0)::send_messages(): m_message_log.size() = 0
5671193.523 WebContent(54430): Document(0x000000011f7c0040)::initialize(0x00000001155c80b8)
5671193.523 WebContent(54430): PageClient(0x00000001155b0040)::initialize_js_console(0x000000011f7c0040): console_client = 0x00000001157f02f0
5671193.529 WebContent(54430): Document(0x000000011f7c07e8)::initialize(0x00000001155c80b8)
5671193.529 WebContent(54430): PageClient(0x00000001155b0040)::initialize_js_console(0x000000011f7c07e8): console_client = 0x00000001157f0448
```

### Without `<select>` element in the document
```
5671130.638 WebContent(54081): Document(0x00000001175f4040)::initialize(0x0000000117450040)
5671130.638 WebContent(54081): PageClient(0x0000000117438040)::initialize_js_console(0x00000001175f4040): console_client = 0x0000000117678040
5671130.639 WebContent(54081): Document(0x00000001175f4040)::make_active()
5671130.639 WebContent(54081): PageClient(0x0000000117438040)::page_did_change_active_document_in_top_level_browsing_context(0x00000001175f4040): m_top_level_document_console_client = 0x0000000117678040
5671130.715 WebContent(54081): Document(0x00000001175f47e8)::initialize(0x00000001174500b8)
5671130.715 WebContent(54081): PageClient(0x0000000117438040)::initialize_js_console(0x00000001175f47e8): console_client = 0x0000000117678198
5671130.715 WebContent(54081): Document(0x00000001175f47e8)::make_active()
5671130.715 WebContent(54081): PageClient(0x0000000117438040)::page_did_change_active_document_in_top_level_browsing_context(0x00000001175f47e8): m_top_level_document_console_client = 0x0000000117678198
5671142.826 WebContent(54081): Document(0x000000011f424040)::initialize(0x0000000117450130)
5671142.826 WebContent(54081): PageClient(0x0000000117438040)::initialize_js_console(0x000000011f424040): console_client = 0x00000001176782f0
5671142.840 WebContent(54081): Document(0x000000011f424040)::make_active()
5671142.840 WebContent(54081): PageClient(0x0000000117438040)::page_did_change_active_document_in_top_level_browsing_context(0x000000011f424040): m_top_level_document_console_client = 0x00000001176782f0
5671142.841 WebContent(54081): Unhandled JavaScript exception: aaa
5671142.841 WebContent(54081): Console(0x000000011756c120)::report_exception(): m_client = 0x00000001176782f0
5671142.841 WebContent(54081): WebContentConsoleClient(0x00000001176782f0)::print_html(<span style="color: red;">Uncaught [Error]: </span>aaa<br>)
5671142.841 WebContent(54081): WebContentConsoleClient(0x00000001176782f0)::print_html(): m_message_log.size() = 1
5671148.064 WebContent(54173): Document(0x0000000115924040)::initialize(0x0000000115780040)
5671148.064 WebContent(54173): PageClient(0x0000000115768040)::initialize_js_console(0x0000000115924040): console_client = 0x00000001159a8040
5671148.065 WebContent(54173): Document(0x0000000115924040)::make_active()
5671148.065 WebContent(54173): PageClient(0x0000000115768040)::page_did_change_active_document_in_top_level_browsing_context(0x0000000115924040): m_top_level_document_console_client = 0x00000001159a8040
5671148.153 WebContent(54173): Document(0x00000001159247e8)::initialize(0x00000001157800b8)
5671148.153 WebContent(54173): PageClient(0x0000000115768040)::initialize_js_console(0x00000001159247e8): console_client = 0x00000001159a8198
5671148.155 WebContent(54173): Document(0x00000001159247e8)::make_active()
5671148.155 WebContent(54173): PageClient(0x0000000115768040)::page_did_change_active_document_in_top_level_browsing_context(0x00000001159247e8): m_top_level_document_console_client = 0x00000001159a8198
5671148.170 WebContent(54081): PageClient(0x0000000117438040)::js_console_request_messages(0): m_top_level_document_console_client = 0x00000001176782f0
5671148.170 WebContent(54081): WebContentConsoleClient(0x00000001176782f0)::send_messages(0)
5671148.170 WebContent(54081): WebContentConsoleClient(0x00000001176782f0)::send_messages(): m_message_log.size() = 1
5671148.170 Ladybird(54066): append_console_output: <span style="color: red;">Uncaught [Error]: </span>aaa<br>
5671148.205 WebContent(54173): Document(0x0000000137110040)::initialize(0x00000001157800b8)
5671148.205 WebContent(54173): PageClient(0x0000000115768040)::initialize_js_console(0x0000000137110040): console_client = 0x00000001159a82f0
5671148.208 WebContent(54173): Document(0x00000001371107e8)::initialize(0x00000001157800b8)
5671148.208 WebContent(54173): PageClient(0x0000000115768040)::initialize_js_console(0x00000001371107e8): console_client = 0x00000001159a8448
5671148.209 WebContent(54173): Document(0x00000001372fc040)::initialize(0x00000001157800b8)
5671148.209 WebContent(54173): PageClient(0x0000000115768040)::initialize_js_console(0x00000001372fc040): console_client = 0x00000001159a85a0
```

**Comment by trflynn89** [comment]
Is that repro right? When I open it with both firefox and ladybird, I don't get any exceptions - and if I print `selector.selectedOptions[0]`, I get an `HTMLOptionElement`. I then don't get a crash in the inspector when running `console.log(123)`.

**Comment by trflynn89** [comment]
ok, I can repro the crash if I remove the 2 `<option>` elements from the HTML:
```html
<div>
    <select id="test-element">
    </select>
</div>
<script>
    let selector = document.getElementById("test-element");
    console.log(selector.selectedOptions[0])
    selector.selectedOptions[0].selected = false
</script>
```

**Comment by nico** [comment]
Fix got reverted in #810; should maybe be reopened.
